src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/GenerateMethodCodeActionResolver.cs

```diff
@@ -132,4 +136,21 @@ private WorkspaceEdit GenerateMethodInCodeBlock(RazorCodeDocument code, Generate
 
         return new WorkspaceEdit() { DocumentChanges = new[] { razorTextDocEdit } };
     }
+
+    private static string PopulateMethodSignature(VersionedDocumentContext documentContext, GenerateMethodCodeActionParams actionParams)
+    {
+        var templateWithMethodSignature = s_generateMethodTemplate.Replace(s_methodName, actionParams.MethodName);
+
+        var returnType = actionParams.IsAsync ? "async System.Threading.Tasks.Task" : "void";
+        templateWithMethodSignature = templateWithMethodSignature.Replace(s_returnType, returnType);
+
+        var eventArgsKey = "Components.EventHandler.EventArgs";
+        var eventTagHelper = documentContext.Project.ProjectWorkspaceState?.TagHelpers
+            .FirstOrDefault(th => th.Name == actionParams.EventName && th.Metadata.ContainsKey(eventArgsKey));
```

Comment:
I took a look at the `TagMatchingRules` for a few of the built in events and I think some default `TagMatchingRules` might be used here since all of the events have `TagMatchingRules` that consists of `TagName = '*'` with `Parent = null` so I am not sure if checking `TagMatchingRules` is helpful here? Though I am not familiar with this enough to know what possible overlaps there could be.. 
For now, I added your suggestion `th.IsEventHandlerTagHelper()` and I think this would reduce the chances for false positives, but I am happy to try make this more rigorous if that is not enough.

Reply:
I think that is fine, we can always follow up if someone finds bugs. I'm guessing custom event handlers is where this could break, but it won't do anything terrible.